File: tests/libs/enums/enums.c
enum color { RED, GREEN = 5, BLUE };

// Takes and returns an enum: exercises enum handling through libffi calls.
enum color brighten(enum color c)
{
    return c + 1;
}

// Returns the integer value of an enum argument.
int as_int(enum color c)
{
    return (int) c;
}

// A global of enum type: exercises enum handling for STT_OBJECT symbols.
enum color favourite = GREEN;

File: tests/py/enums/enums.py
from pyc_testlib import add_libs
add_libs("enums")
from elflib import enums as m

# Enum global is read as its underlying integer value (GREEN == 5).
print(m.favourite)
# Enum passed as argument and returned (brighten(RED) -> 1).
print(m.brighten(0))
# Enum returned then passed back in as an argument.
print(m.as_int(m.favourite))
# Larger enum value round-trips through a libffi call.
print(m.brighten(m.favourite))
